test/should_fail/map_pattern.erl

```diff
@@ -9,3 +9,8 @@ f(#{bepa := Bepa}) ->
 
 -spec badkey(#{apa => atom()}) -> ok.
 badkey(#{bepa := _Bepa}) -> ok.
+
+-spec map_term(term() | #{atom() => atom()} | term()) -> any().
```

Comment:
Shouldn't the spec be:
`-spec map_term(term() | #{atom() => atom() | term()}) -> any().`

I'm confused about what the spec even means. Either the parameter is a `term()` or it's a `#{atom() => atom() | term()}`, but `#{atom() => atom() | term()}` already is a `term()` so that's superfluous, right?

Reply:
ops, I was just trying to provoke some other error case in `add_type_pat` and I forgot the type here. the type is normalized anyway so I guess it should be just `term()`.